src/compiler/debug.ts

```diff
@@ -781,5 +781,36 @@ namespace ts {
             }
             return result;
         }
+
+        type DebugType = Type & { __debugTypeToString(): string }; // eslint-disable-line @typescript-eslint/naming-convention

```

Comment:
Can this remain exported? I can exported it when I need it but it's kind of cool to write code like:

```ts
if (Debug.isDebugging) {
    console.log(`${(type as DebugType}.__debugTypeToString()`)
}
```